js/src/ScatterBase.ts

```diff
@@ -26,13 +26,17 @@ export abstract class ScatterBase extends mark.Mark {
     render() {
         const base_creation_promise = super.render();
 
+        var that = this;
+        // Warning: arrow functions actually breaks the drag; that should be fixed
+        // with generics from the latest definitions of d3-selection and d3-drag
+        // from @types
         this.drag_listener = d3.drag()
-          .subject((d: any) => {
-              return {x: this.x_scale.scale(d.x), y: this.y_scale.scale(d.y)};
+          .subject(function(d: any) {
```

Comment:
an arrow function is actually okay here, right? However, +1 for keeping it `function` for consistency with the below calls.

Reply:
Indeed this one could be replaced with an arrow function.